docs/en/sql-reference/functions/array-functions.md

```diff
@@ -2118,6 +2118,113 @@ Result:
 └─────────────────────┘
 ```
 
+
+## randomSampleFromArray
+
+Returns a random sample from an input array. The number of elements in the sample is determined by the `sampleSize` argument. The function ensures that each element in the input array has an equal chance of being included in the sample.
+
+**Syntax**
+
+```sql
+randomSampleFromArray(arr, sampleSize)
+```
+
+**Arguments**
+
+- `arr` — The input array from which to sample elements, including nested arrays.
+- `sampleSize` — The number of elements to include in the random sample.
+
+**Returned value**
+
+- An array containing a random sample of elements from the input array.
+
+**Examples**
+
+Query:
+
+```sql
+SELECT randomSampleFromArray([1, 2, 3, 4, 5, 6], 3) as res;
+```
+
+Result:
+```
+┌─res─────┐
+│ [2,3,1] │
+└─────────┘
+or
+┌─res─────┐
+│ [5,2,6] │
+└─────────┘
+```
+
+Query:
+
+```sql
+SELECT randomSampleFromArray(['apple', 'banana', 'cherry', 'date'], 2) as res;
+```
+
+Result:
+```
+┌─res────────────────┐
+│ ['apple','banana'] │
+└────────────────────┘
+or
+┌─res────────────────┐
+│ ['banana','apple'] │
+└────────────────────┘
+or
+┌─res───────────────┐
+│ ['cherry','date'] │
+└───────────────────┘
+```
+
+Query:
+
+```sql
+SELECT randomSampleFromArray([[1, 2], [3, 4], [5, 6]], 2) as res;
+```
+
+Result:
+```
+┌─res───────────┐
+│ [[1,2],[5,6]] │
+└───────────────┘
+or
+┌─res───────────┐
+│ [[3,4],[5,6]] │
+└───────────────┘
+```
+
+Query:
+
+```sql
+SELECT randomSampleFromArray([1, 2, 3, 4, 5], 0) as res;
+```
+
+Result:
+```
+TODO: FIX ME
+```
+
+Query:
+
+```sql
```

Comment:
Let's omit the example for this special case, it's clear what happens anyways.

Reply:
Ah, I missed that the results are in different order, potentially. Then keep the example - thanks.